src/mavsdk/plugins/camera/camera_definition.cpp

```diff
@@ -110,18 +110,21 @@ bool CameraDefinition::parse_xml()
             return false;
         }
 
-        const char* type_str = e_parameter->Attribute("type");
-        if (!type_str) {
+        const char* type_str_res = e_parameter->Attribute("type");
+        // The check for null pointer is required before the creation of std::string.
```

Comment:
```suggestion
```

Reply:
I thought this comment would be good to have, since if someone would remove the `if` or somehow change it, the runtime exception from creating a std::string from NULL could happen.

Reply:
The comment was good for me to review actually but I would expect someone to work on this code to see all the examples of checking for non-null and therefore not require it but it's fine, I'm happy to leave it.

Reply:
Good point, I suppose no one would delete a null check just for fun. I'll remove it.

Reply:
Removed in [12a8070](https://github.com/mavlink/MAVSDK/pull/2056/commits/12a807014c365b61db423a6ff6ea94348167e060)